File: engine/engine-model/src/main/scala/tools/packet/inventory/InventoryFull.scala
package tools.packet.inventory

import java.util.Collections

import client.inventory.ModifyInventory
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class InventoryFull() extends PacketInput {
  //noinspection MutatorLikeMethodIsParameterless
  def updateTick: Boolean = true

  def modifications: java.util.List[ModifyInventory] = Collections.emptyList()

  override def opcode(): SendOpcode = SendOpcode.INVENTORY_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/inventory/InventoryFull.java
package tools.packet.inventory;

import java.util.Collections;
import java.util.List;

import client.inventory.ModifyInventory;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record InventoryFull(Boolean updateTick, List<ModifyInventory> modifications) implements PacketInput {
   public InventoryFull() {
      this(true, Collections.emptyList());
   }

   @Override
   public SendOpcode opcode() {
      return SendOpcode.INVENTORY_OPERATION;
   }
}